src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
```

Comment:
The import statement for NoSuchElementException is correct, but make sure to use it consistently across your code. Currently, you are using ArrayListIndexOutOfBoundsException, which is not a standard Java exception. Consider using IndexOutOfBoundsException instead.